test/common/ftr_provider_context.d.ts

```diff
@@ -17,7 +17,8 @@
  * under the License.
  */
 
-export { KibanaServerProvider } from './kibana_server';
-export { EsProvider } from './es';
-export { EsArchiverProvider } from './es_archiver';
-export { RetryProvider } from './retry';
+import { GenericFtrProviderContext } from '@kbn/test/types/ftr';
+
+import { services } from './services';
+
+export type FtrProviderContext = GenericFtrProviderContext<typeof services, {}>;
```

Comment:
Probably could take more consistency name, like `ProviderContext`, `Context` or `TestContext`

Reply:
Well, the context is defined by the functional test runner (Ftr), and there are a lot of providers out there, and even more contexts. It's also not just for tests, it's for services, page objects, and test providers.